core/src/HIP/Kokkos_HIP_Instance.cpp

```diff
@@ -287,6 +290,15 @@ void HIPInternal::initialize(int hip_device_id, hipStream_t stream) {
 
   // Init the array for used for arbitrarily sized atomics
   if (m_stream == nullptr) ::Kokkos::Impl::initialize_host_hip_lock_arrays();
+
+  // Allocate a staging buffer for constant mem in pinned host memory
+  // and an event to avoid overwriting driver for previous kernel launches
+  if (m_stream == nullptr) {
+    HIP_SAFE_CALL(hipHostMalloc((void **)&constantMemHostStaging,
+                                HIPTraits::ConstantMemoryUsage));
```

Comment:
I know you copied that from Cuda but wondering now whether that memory should be tracked via `HIPHostPinnedSpace::allocate`.  @crtrott what do you think?

Reply:
probably.